src/autonomous/AutonomousTask.cpp

```diff
@@ -0,0 +1,57 @@
+#include "AutonomousTask.h"
+
+#include "../Constants.h"
+#include "../Globals.h"
+#include "../commands/DriveToWaypointCommand.h"
+#include "../world_interface/world_interface.h"
+
+namespace autonomous {
+
+AutonomousTask::AutonomousTask(){};
+
+AutonomousTask::~AutonomousTask() {
+	if (_autonomous_task_thread.joinable()) {
+		_autonomous_task_thread.join();
+	}
+}
+
+void AutonomousTask::start(const navtypes::pose_t& waypointCoords) {
+	setWaypoint(waypointCoords);
+	_kill_called = false;
+	_autonomous_task_thread = std::thread(&autonomous::AutonomousTask::navigate, this);
+}
+
+void AutonomousTask::navigate() {
+	commands::DriveToWaypointCommand cmd(_waypoint_coords, THETA_KP, DRIVE_VEL, SLOW_DRIVE_VEL,
+										 DONE_THRESHOLD, CLOSE_TO_TARGET_DUR_VAL);
+
+	while (!cmd.isDone()) {
+		auto latestGPS = robot::readGPS();
+		if (latestGPS.isFresh(2000ms)) {
+			cmd.setState(latestGPS, robot::types::dataclock::now());
+			commands::command_t output = cmd.getOutput();
+			robot::setCmdVel(output.thetaVel, output.xVel);
+		}
+		std::unique_lock autonomousTaskLock(_autonomous_task_mutex);
+		// Wait 1000ms or return out of while loop if kill called
+		if (_autonomous_task_cv.wait_until(autonomousTaskLock,
+										   std::chrono::system_clock::now() + 1000ms,
+										   [&] { return _kill_called; })) {
+			return;
+		}
+	}
+}
+
+void AutonomousTask::kill() {
+	_kill_called = true;
```

Comment:
`_kill_called` should be protected with a mutex. Also, after tripping the flag, you should notify the CV. Something like:

```C++
{
    std::lock_guard lock(autonomous_mutex);
    _kill_called = true;
}
_autonomous_task_cv.notify_all();
```